src/test/java/mate/academy/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,169 @@
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    private static final String ID = "/{id}";
+    private static final String TITLE_PARAM_NAME = "title";
+    private static final String AUTHOR_PARAM_NAME = "author";
+    private static final String ISBN_PARAM_NAME = "isbn";
+    private static final String PRICE_PARAM_NAME = "price";
+    private static final String BASE_URL = "/books";
+    private static final String SEARCH_URL = BASE_URL + "/search";
+    private static final String ADMIN_ROLE = "ADMIN";
+    private static final String USER_ROLE = "USER";
+    private static final String VALID_BOOK_TITLE = "Kobzar";
+    private static final String VALID_BOOK_AUTHOR = "Taras Shevchenko";
+    private static final String VALID_BOOK_ISBN = "978-0-7847-5628-7";
+    private static final BigDecimal VALID_BOOK_PRICE = BigDecimal.valueOf(14.99);
+    private static final Set<Long> VALID_CATEGORY_ID = Set.of(1L);
+    private static final Long VALID_BOOK_ID = 1L;
+    private static final String TITLE_0_EXPRESSION = "$[0].title";
+    private static final String TITLE_EXPRESSION = "$.title";
+
+    private static MockMvc mockMvc;
+
+    @MockBean
+    private BookService bookService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private BookDto bookDto;
+    private CreateBookRequestDto createBookDto;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext context) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(context)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @BeforeEach
+    void setup() {
+        bookDto = new BookDto();
+        bookDto.setId(VALID_BOOK_ID);
+        bookDto.setTitle(VALID_BOOK_TITLE);
+        bookDto.setAuthor(VALID_BOOK_AUTHOR);
+        bookDto.setIsbn(VALID_BOOK_ISBN);
+        bookDto.setPrice(VALID_BOOK_PRICE);
+
+        createBookDto = new CreateBookRequestDto();
+        createBookDto.setTitle(VALID_BOOK_TITLE);
+        createBookDto.setAuthor(VALID_BOOK_AUTHOR);
+        createBookDto.setIsbn(VALID_BOOK_ISBN);
+        createBookDto.setCategoryIds(VALID_CATEGORY_ID);
+        createBookDto.setPrice(VALID_BOOK_PRICE);
+    }
+
+    @Test
+    @WithMockUser(roles = {USER_ROLE, ADMIN_ROLE})
+    @DisplayName("Get all books (valid request)")
+    void getAllBooks_ValidRequest_ReturnsListOfBooks() throws Exception {
+        when(bookService.findAll(any(Pageable.class))).thenReturn(
+                Collections.singletonList(bookDto));
+
+        mockMvc.perform(get(BASE_URL))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath(TITLE_0_EXPRESSION).value(VALID_BOOK_TITLE));
```

Comment:
Better to check full response object (check other cases as well)

Reply:
not resolved